docs/separate-benchmark-source-set.md

```diff
@@ -0,0 +1,112 @@
+    In this example, we're associating the new compilation `benchmark` with the `main` compilation of the `jvm` target. 
+    This association allows the benchmark compilation to access the internal API of the main compilation, 
+    which is particularly useful when benchmarks need to measure the performance of specific components 
+    or functionalities within the main codebase.
+
+    ```kotlin
+    // build.gradle.kts
+    kotlin { 
+        jvm { 
+            compilations.create('benchmark') {
+                associateWith(compilations.main)
+            }
+        }
+    }
+    ```
+
+2. **Register Benchmark Compilation**
+
+    Register your new benchmark compilation using its default source set name. 
+    In this instance, `jvmBenchmark` is the name of the default source set of the `benchmark` compilation.
+
+    ```kotlin
+    // build.gradle.kts
+    benchmark {
+        targets {
+            register("jvmBenchmark")
+        }
+    }
+    ```
+
+3. **Add Benchmarks**
+
+    Place your benchmark code into the default source set of the benchmark compilation. 
+    The default source set can also depend on other source sets with benchmarks. 
+    This way you can share benchmarks between multiple benchmark compilations. 
+    Refer to our [writing benchmarks guide](docs/writing-benchmarks.md) for a comprehensive guide on writing benchmarks.
+
+For additional information, refer to the [Kotlin documentation on creating a custom compilation](https://kotlinlang.org/docs/multiplatform-configure-compilations.html#create-a-custom-compilation).
+and the [documentation on associating compiler tasks](https://kotlinlang.org/docs/gradle-configure-project.html#associate-compiler-tasks).
+[Here is a sample Kotlin Multiplatform project](/examples/kotlin-multiplatform) with a separate compilation for benchmarks.
+
+### Kotlin/JVM Project
+
+Set up a separate benchmark source set by following these simple steps:
+
+1. **Define Source Set**
+
+    Begin by defining a new source set. We'll use `benchmark` as the name for the source set.
+
+    ```kotlin
+    // build.gradle.kts
+    sourceSets {
+        create("benchmark")
+    }
+    ```
+
+2. **Propagate Dependencies**
+
+    Next, propagate dependencies and output from the `main` source set to your `benchmark` source set. 
+    This ensures the `benchmark` source set has access to classes and resources from the `main` source set.
+
+    ```kotlin
+    // build.gradle.kts
+    dependencies {
+        add("benchmarkImplementation", sourceSets.main.get().output + sourceSets.main.get().runtimeClasspath)
+    }
+    ```
+
+    You can also add output and `compileClasspath` from `sourceSets.test` in the same way 
+    if you wish to reuse some of the test infrastructure.
+
+3. **Register Benchmark Source Set**
+
+    Register your benchmark source set. This informs the kotlinx-benchmark tool 
+    that benchmarks reside within this source set and need to be executed accordingly.
```

Comment:
```suggestion
    about benchmarks reside within this source set and need to be executed accordingly.
```

Reply:
I am not sure if the change is grammatically correct.